cmd/deployIntegrationArtifact.go

```diff
@@ -0,0 +1,158 @@
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runDeployIntegrationArtifact(&config, telemetryData, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runDeployIntegrationArtifact(config *deployIntegrationArtifactOptions, telemetryData *telemetry.CustomData, httpClient piperhttp.Sender) error {
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return errors.Wrap(cookieErr, "creating a cookie jar failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+
+	deployURL := config.Host +
+		"/api/v1/DeployIntegrationDesigntimeArtifact?Id=" + "'" + config.IntegrationFlowID +
+		"'" + "&Version=" + "'" + config.IntegrationFlowVersion + "'"
+
+	finalResult, err := getBearerTokenforDeployIntegrationArtifactCall(config, httpClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch Bearer Token")
+	}
+	clientOptions.Token = "Bearer " + finalResult
+	httpClient.SetOptions(clientOptions)
+
+	deployResp, httpErr := httpClient.SendRequest("POST", deployURL, nil, header, nil)
+
+	defer func() {
+		if deployResp != nil && deployResp.Body != nil {
+			deployResp.Body.Close()
+		}
+	}()
+
+	if deployResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
+	}
+
+	if deployResp.StatusCode == 202 {
+		log.Entry().
+			WithField("IntegrationFlowID", config.IntegrationFlowID).
+			Info("successfully deployed in to CPI runtime")
+		return nil
+	}
+
+	log.Entry().Errorf("a HTTP error occurred! Response Status Code: %v", deployResp.StatusCode)
+	return errors.Wrap(httpErr, "Deploying the integration flow failed")
+}
+
+func getBearerTokenforDeployIntegrationArtifactCall(config *deployIntegrationArtifactOptions, httpClient piperhttp.Sender) (string, error) {
+
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return "", errors.Wrap(cookieErr, "creating a cookie jar failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+	tokenURL := config.OAuthTokenProviderURL + "?grant_type=client_credentials"
+	method := "POST"
+	resp, httpErr := httpClient.SendRequest(method, tokenURL, nil, header, nil)
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+	// for supporting tests
+	if resp.Header.Get("Authorization") != "" {
+		result := resp.Header.Get("Authorization")
+		return result, nil
+	}
+
+	if resp == nil {
+		return "", errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
+	}
+	if resp.StatusCode != 200 {
```

Comment:
Maybe it would make sense to put this above before returing the `Authorization` header since it does not make sense most likely to expect/return such a header in case of an unexpected status code.